File: python/svix/models/ingest_endpoint_update.py
# this file is @generated
import typing as t

from .common import BaseModel


class IngestEndpointUpdate(BaseModel):
    description: t.Optional[str] = None

    disabled: t.Optional[bool] = None

    metadata: t.Optional[t.Dict[str, str]] = None

    rate_limit: t.Optional[int] = None

    uid: t.Optional[str] = None
    """Optional unique identifier for the endpoint."""

    url: str

File: kotlin/lib/src/main/kotlin/models/IngestEndpointUpdate.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class IngestEndpointUpdate(
    val description: String? = null,
    val disabled: Boolean? = null,
    val metadata: Map<String, String>? = null,
    val rateLimit: UShort? = null,
    /** Optional unique identifier for the endpoint. */
    val uid: String? = null,
    val url: String,
)
